menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -102,4 +102,10 @@ class DatasetService @Autowired() (datasetMongoRepository: DatasetMongoRepositor
     }
   }
 
+  override def importSingleItem(item: Dataset, username: String): Future[Option[Dataset]] = {
+    getLatestVersionValue(item.name).flatMap {
+      case Some(version) => update(username, item.copy(version = version))
+      case None => super.create(item.copy(version = 1), username)
+    }
+  }
```

Comment:
This here bugs me. This is something that is in `DatasetService`, `SchemaService` and `MappingTableService` but cannot be abstracted to `VersionedModelService` because of the usage of method `copy` which need implementation of `C` like `Dataset`.  Not sure what to do.

Any ideas on how to do it?